### ISSUE TITLE
Allow specifying initial directory via environment variable

### ISSUE THREAD

**Issue by ShellCode33** [keepassxreboot/keepassxc#9182]
## Summary

By default the home folder is open:

![image](https://user-images.githubusercontent.com/8455652/221898476-13b2aea4-fe03-4451-88bc-558b4b51740e.png)

While this is a sensible default, it would be great to be able to customise which folder to open.

We could customise this using an environment variable.

## Context

I'm using AppArmor and I'd like to confine KeepassXC to be able to access the bare minimum. Why ? Because it is able to interact with other processes (ssh-agent, browser extension, ...) and therefore I consider its attack surface big enough to be willing to confine it.

Here's the AppArmor rules I'm using to restrict its access to files in my home folder:

```
owner @{HOME}/@{XDG_DATA_DIR}/keepassxc/ rwl,
owner @{HOME}/@{XDG_DATA_DIR}/keepassxc/** rwl,
owner @{HOME}/@{XDG_CONFIG_DIR}/keepassxc/** rwlk,
```

It is basically unable to open anything else than `~/.config/keepassxc/*` and `~/.local/share/keepassxc/*` in my home directory.
Thing is, I cannot browse to `~/.local/share/keepassxc/*` (where my database is located) using the file chooser because it doesn't have the right to list directories in `~`, `~/.local/` and `~/.local/share/`.

This is probably very specific and does not concern many people, but KeepassXC being a security tool, I think it makes sense to consider such use cases.

Thanks for reading me !

**Comment by droidmonkey** [comment]
This isn't our problem to solve, you need to provide directory list capability to traverse folders. Or use a better folder like `~/kpxc`. We already save the last used folder for opening a database. 

If you are so concerned about ssh agent and browser extension then just disable there use in the settings. In my opinion you are going to great lengths at sandboxing for zero gain. If you don't trust our software to operate on your system, then why are you using it to store your passwords?

**Comment by ShellCode33** [comment]
> This isn't our problem to solve

I don't know if we can call it a problem, but it is definitely something you and only you can handle.

> you need to provide directory list capability to traverse folders

Yeah so basically what you are suggesting is to lower my security expectations because you don't want to implement something as simple as an environment variable... "providing directory list capability to traverse folders" is exactly what I want to avoid

> ssh agent and browser extension then just disable there use in the settings

Well I'd like to keep using these features, they are cool :)

> If you don't trust our software to operate on your system, then why are you using it to store your passwords

I do trust your software, otherwise I wouldn't be using it, but that's irrelevant. I know for a fact that even the most well written programs have holes in them, no software is perfect. That's why we invented defense in depth, to have multiple layers of security, so that even if one layer breaks, the others protect you from the threat. That includes sandboxing mecanisms.


If I provide a PR for it, will you accept it ?

**Comment by droidmonkey** [comment]
I would not accept a pr for this very specific feature. Can you even open the folder by directly typing in the path in the open file dialog? You do this once and it will be saved in the config.

**Comment by ShellCode33** [comment]
> Can you even open the folder by directly typing in the path in the open file dialog

Yes I can, but I plan to give my setup to others, and it would be very bad UX to require such a thing. I need to change the default behavior so that it's usable from a fresh setup by anyone.

![image](https://user-images.githubusercontent.com/8455652/222009306-540ab8d1-299e-4338-9a70-9c340d9bf3f0.png)

-------

> You do this once and it will be saved in the config

No it wont :
![image](https://user-images.githubusercontent.com/8455652/222008776-696c0359-fb4f-459f-a4e5-699097216ad1.png)

I don't need any cache, I just wish to be able to open an alternative default directory where the database is. And apparently I'm not the only one who would appreciate such a feature #2750

**Comment by droidmonkey** [comment]
You need to allow writing the local config file or change it's location with an env var.....

I'm not sure who would want their database file stored in an obtuse location like ~/.local/share/keepassxc. That also makes it impossible to do version control and backups of your database. None of this really makes sense to me. Do you prevent your browser from accessing every folder in your home directory?

**Comment by ShellCode33** [comment]
This is intended behaviour. Even if I allow it to write its config, it wont solve the fact that :

> I plan to give my setup to others, and it would be very bad UX to require such a thing

I really don't understand why you are so against something as simple as getting the default folder from an environment variables, this is 2 or 3 lines of code at most

**Comment by droidmonkey** [comment]
I'm not really against the code, I'm against this whole construct. It just doesn't make sense to me. You invent a problem and then ask us to incorporate code to solve your invented problem.

**Comment by ShellCode33** [comment]
I'm not inventing any problem, it's just that I care about security and that I try to lower my applications capabilities as much as possible. This is called the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege) and is security 101.

Beside security, there are many use cases that would benefic from such a feature, and I'm sure that even if people are too lazy to open an issue here on Github, many people would actually use it. Imagine working at a company where you have network shares on which there are multiple KDBX databases, and each database is shared between only a few people (let's say the company has a Twitter account but also has 5 community managers for example). Don't you think it would be handy to be able to set a default path instead of requiring the end users to go to `Z:/The/Location/Of/Keepass/XC/Databases/Is/Over/There/Oh/No/We/Moved/It/Yesterday/Its/Actually/There/` ? I do. I know you will tell me "yeah but there is the recently used DB so thats ok", except it doesnt answer the same need. What about new people starting at this company ? What about the sysadmin changing the location of the DBs on the share ?

This is not some obscur feature I'm asking, this is something very basic that could be useful for many people, way more useful than some other features you already accepted to add:

![image](https://user-images.githubusercontent.com/8455652/222021727-7c1d157e-a36c-4d91-a4d0-231be1223fa0.png)


Plus it's very simple to do and won't clutter your code.

**Comment by droidmonkey** [comment]
To be fair your choice of directory invented the problem, I wasn't referring to the use of apparmor.

**Comment by ShellCode33** [comment]
To be fair this is not "my choice of directory", I'm just following the [FreeDesktop specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) which states :

> $XDG_DATA_HOME defines the base directory relative to which user-specific data files should be stored. If $XDG_DATA_HOME is either not set or empty, **a default equal to $HOME/.local/share should be used**.

Thanks for your reconsideration ;)

---------

EDIT:

I didn't see your edit:

> Do you prevent your browser from accessing every folder in your home directory

I do, my browser has read only access to `~/share` (to upload files) and write only access to `~/downloads`

**Comment by ShellCode33** [comment]
See #9192 :)
